### PR TITLE
fix oracle contract voter sort

### PR DESCRIPTION
**Pull request by lorenzo-dev1**
The oracle contract `_votersSort` function sorts the voters in descending address order, but the `_updateVotingInfo` function expects them sorted in ascending order. This PR fixes the sorting algo to sort ascending and adds some tests for it.

Tests were not failing before because the addresses generated by the truffle contract tests were not hitting this edge case.

### CODE REVIEW

File: autonity/solidity/contracts/Oracle.sol [hunk 1]
```diff
@@ -280,14 +280,13 @@ contract Oracle is IOracle {
     */
     function _votersSort(address[] memory _voters, int _low, int _high)
     internal pure {
+        if (_low >= _high) return;
         int _i = _low;
         int _j = _high;
-        if (_i == _j) return;
```
Comment (yazzaoui) [inline autonity/solidity/contracts/Oracle.sol:285]: Wouldn't that make the sorting unstable?

**Review by yazzaoui — APPROVED** [review]
LGTM

**Review by vpiyush — APPROVED** [review]
LGTM!
